components/script/script_thread.rs

```diff
@@ -808,12 +808,13 @@ impl ScriptThread {
     }
 
     pub fn mark_document_with_no_blocked_loads(doc: &Document) {
-        SCRIPT_THREAD_ROOT.with(|root| {
-            let script_thread = unsafe { &*root.get().unwrap() };
-            script_thread
-                .docs_with_no_blocking_loads
-                .borrow_mut()
-                .insert(Dom::from_ref(doc));
+        SCRIPT_THREAD_ROOT.with(|root| unsafe {
+            if let Some(script_thread) = root.get() {
+                (*script_thread)
+                    .docs_with_no_blocking_loads
+                    .borrow_mut()
+                    .insert(Dom::from_ref(doc));
+            }
```

Comment:
Let's scope the unsafe block a little bit narrower:
```rust
SCRIPT_THREAD_ROOT.with(|root| {
    if let Some(script_thread) = root.get() {
        let script_thread = unsafe { &*script_thread };
        script_thread
            .docs_with_no_blocking_loads
            .borrow_mut()
            .insert(Dom::from_ref(doc));
    }
});
```

Reply:
thank you! I didn't know I could deref the bound value in an if let like that.